test/types/function.test-d.ts

```diff
@@ -0,0 +1,29 @@
+import { expectError, expectNotType, expectType } from 'tsd';
+import App from '../../src/App';
+import type { FunctionCompleteFn, FunctionFailFn } from '../../src/CustomFunction';
+import type { FunctionInputs } from '../../src/types';
+import type { AckFn } from '../../src/types/utilities';
+
+const app = new App({ token: 'TOKEN', signingSecret: 'Signing Secret' });
+
+expectError(
+  app.function('callback', async ({ ack }) => {
+    expectNotType<AckFn<void>>(ack);
+  }),
+);
+
+expectError(
+  app.function('callback', { autoAcknowledge: true }, async ({ ack }) => {
+    expectNotType<AckFn<void>>(ack);
+  }),
+);
+
+app.function('callback', { autoAcknowledge: false }, async ({ ack }) => {
```

Comment:
Perfect! Just need a comment above this describing the test and its purpose. You and I understand what the point of this test is since we are familiar with this system, but perhaps an outside contributor will not. The comment is helpful to describe _why_ we are testing what we are testing. Usually test names in the form "it should [not] do something when [certain condition]." work pretty well to communicate that. The other added benefit from this is that by simply reading the tests, a person with zero context can readily learn _how_ bolt should work and what it provides.